src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -13,50 +13,34 @@ namespace System.Net
 {
     internal static partial class NameResolutionPal
     {
-        private static volatile bool s_initialized;
-        private static readonly object s_initializedLock = new object();
+        private static volatile int s_getAddrInfoExSupported;
 
-        private static readonly unsafe Interop.Winsock.LPLOOKUPSERVICE_COMPLETION_ROUTINE s_getAddrInfoExCallback = GetAddressInfoExCallback;
-        private static bool s_getAddrInfoExSupported;
-
-        public static void EnsureSocketsAreInitialized()
+        public static bool SupportsGetAddrInfoAsync
         {
-            if (!s_initialized)
+            get
             {
-                InitializeSockets();
-            }
+                if (s_getAddrInfoExSupported == 0)
+                    Initialize();
+                return s_getAddrInfoExSupported == 1;
```

Comment:
Might these two volatile reads on arm be measurable?  Wondering if it'd be better as:
```C#
int supported = s_getAddrInfoExSupported;
if (supported == 0)
{
    Initialize();
    supported = s_getAddrInfoExSupported;
}
return supported;
```
Or is that going to prohibit this from being inlined and end up being net worse?